### PR TITLE
[PR-20964] Add ability to pass access_token as parameter

### PR DESCRIPTION
**Pull request by bohdan-perekhrest**
[Product changes](talkable/talkable#9880)

[PR-20964]: https://talkable.atlassian.net/browse/PR-20964?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by zhuravel** [comment]
@bohdan-perekhrest Please mention the PR that will use this code in this PR’s description.

**Review by bohdan-perekhrest — COMMENTED** [review]
@zhuravel Please take a look at first commit and last commit, how it will be better?

File: lib/exact_target_sdk/client.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ module ExactTargetSDK
 # the guide linked above.
 class Client
 
+  attr_reader :config
```
Comment (zhuravel) [inline lib/exact_target_sdk/client.rb]:
```suggestion
  attr_reader :config

```

File: lib/exact_target_sdk/client.rb [hunk 2]
```diff
@@ -320,14 +321,16 @@ def execute_request(method)
   # https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/access-token-s2s.html
   # https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/authenticate-soap-api.html#authenticate-with-an-access-token
   def access_token_authorization?
-    config[:endpoint].include?(".auth.marketingcloudapis.com")
+    config[:endpoint].include?(".auth.marketingcloudapis.com") || !config[:access_token].nil?
```
Comment (zhuravel) [inline lib/exact_target_sdk/client.rb]: Why change this method?

**Review by zhuravel — APPROVED** [review]

**Pull request merged**